src/apps/nagle_rcv.c

```diff
@@ -152,31 +153,30 @@ main(int argc, char *argv[])
 	}
 	
 	/* Set some basic values which depend on the address family. */
-#if TEST_V6
-        hst = gethostbyname2(local_host, AF_INET6);
-        if (hst == NULL || hst->h_length < 1) {
-                fprintf(stderr, "%s: bad hostname: %s\n", argv[0], local_host);
+        if (strcmp(local_host, "0") == 0)
+                local_host = "0.0.0.0";
+
+        memset(port_buffer, 0, sizeof(port_buffer));
+        snprintf(port_buffer, 10, "%d", local_port);
+        error = getaddrinfo(local_host, port_buffer, NULL, &hst_res);
+        if (error) {
+                fprintf(stderr, "%s: getaddrinfo failed: %s\n", argv[0], local_host);
                 exit(1);
         }
-	pf_class = PF_INET6;
 
-        host.v6.sin6_family = AF_INET6;
-        memcpy(&host.v6.sin_addr, hst->h_addr_list[0], hst->h_length);
-        host.v6.sin6_port = htons(local_port);
-
-#else
-	hst = gethostbyname(local_host);
-        if (hst == NULL || hst->h_length < 1) {
-                fprintf(stderr, "%s: bad hostname: %s\n", argv[0], local_host);
+        pf_class = hst_res->ai_family;
+        switch (pf_class) {
+        case AF_INET:
+        case AF_INET6:
+                memcpy(&host.sa, hst_res->ai_addr, hst_res->ai_addrlen);
+                break;
+        default:
+                fprintf(stderr, "Invalid address type.\n");
                 exit(1);
+                break;
         }
-	pf_class = PF_INET;
-
-        host.v4.sin_family = AF_INET;
-        memcpy(&host.v4.sin_addr, hst->h_addr_list[0], hst->h_length);
-        host.v4.sin_port = htons(local_port);
 
-#endif /* TEST_V6 */
+        free(hst_res);
```

Comment:
it's better to use freeaddrinfo() instead of free() to free the results of getaddrinfo()